iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveTableUtil.java

```diff
@@ -169,7 +169,10 @@ private static List<DataFile> getDataFiles(RemoteIterator<LocatedFileStatus> fil
           dataFile -> {
             StructLike structLike = dataFile.partition();
             for (int pos : stringFields) {
-              structLike.set(pos, FileUtils.unescapePathName(structLike.get(pos, String.class)));
+              String val = structLike.get(pos, String.class);
```

Comment:
@vikramahuja1001 i think we can simply revert HIVE-27367,  https://github.com/apache/iceberg/pull/7744 fixed the issue. Could you please try it out.

Reply:
Thanks for the review @deniskuzZ . I will check the same and revert back.

Reply:
@deniskuzZ , simple reverting the code part of HIVE-27367 works in this case. I have reverted the code but have kept the test case.
Also added new tests as well as a part of this PR.